src/Twilio/Security/RequestValidator.cs

```diff
@@ -81,27 +129,16 @@ public bool ValidateBody(string rawBody, string expected)
             {
                 dict.Add(k, col[k]);
             }
+
             return dict;
         }
 
-        private string GetValidationSignature(string url, IDictionary<string, string> parameters)
+        private string GetValidationSignature(string urlWithParameters)
         {
-            var b = new StringBuilder(url);
-            if (parameters != null)
-            {
-                var sortedKeys = new List<string>(parameters.Keys);
-                sortedKeys.Sort(StringComparer.Ordinal);
-
-                foreach (var key in sortedKeys)
-                {
-                    b.Append(key).Append(parameters[key] ?? "");
-                }
-            }
-
-            var hash = _hmac.ComputeHash(Encoding.UTF8.GetBytes(b.ToString()));
+            byte[] hash = _hmac.ComputeHash(Encoding.UTF8.GetBytes(urlWithParameters));
             return Convert.ToBase64String(hash);
-        }
-
+        }
+
         private static bool SecureCompare(string a, string b)
```

Comment:
Unrelated to this pull request, but if you want this to be "secure" then you should apply `[MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]` to this method. It might result in a trivial hit to perf, but, the point of secure comparisons is that they are immune from any optimizations.

Reply:
Interesting! Is there a secure compare method built into .NET? Is this usually how this type of compare is written?